### PR TITLE
Add cmake package targets

### PR DESCRIPTION
**Pull request by bmarques1995**
These changes will let the user call the command `find_package(libgit2)` when the library is installed, this can be use as a 
guarantee that the package is properly installed.

### CODE REVIEW

**Comment by bmarques1995** [comment]
> Any progress on this PR? This's really helpful to someone who wants to get libgit2 easily used in a CMake based project.

The task is ready, but need to be accepted

File: src/cli/CMakeLists.txt [hunk 1]
```diff
@@ -53,4 +53,4 @@ if(MSVC_IDE)
 	set_source_files_properties(win32/precompiled.c COMPILE_FLAGS "/Ycprecompiled.h")
 endif()
 
-install(TARGETS git2_cli RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS git2_cli EXPORT ${LIBGIT2_TARGETS_EXPORT_NAME} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
```
Comment (ethomson) [inline src/cli/CMakeLists.txt:56]: Curious why we'd export the CLI targets? How would somebody make use of this and why would they want it? 🤔

**Comment by ethomson** [comment]
Thanks for your patience. Every time I look at cmake, my eyes glaze over, and I need to go lie down. Just one question to start with about the CLI.

File: src/cli/CMakeLists.txt (hunk 1 above)
Comment (bmarques1995) [reply src/cli/CMakeLists.txt:56]: You are correct, for the git cli, the export is unnecessary.

**Comment by ethomson** [comment]
Thanks for the fix! This will land in v1.9.

**Pull request merged**